### PR TITLE
{bio}[foss/2020b] BOLT-LMM v2.3.4

### PR DESCRIPTION
**Pull request by btlindert**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-fix-makefile.patch [hunk 1]
```diff
@@ -0,0 +1,38 @@
+diff -ru BOLT-LMM_v2.3.4.orig/src/Makefile BOLT-LMM_v2.3.4/src/Makefile
```
Comment (verdurin) [inline easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-fix-makefile.patch:1]: @btlindert please add a comment here explaining the purpose of the patch.

File: easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-2.3.4-foss-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+easyblock = 'MakeCp'
+
+name = 'BOLT-LMM'
+version = '2.3.4'
+
+homepage = 'https://storage.googleapis.com/broad-alkesgroup-public/BOLT-LMM/BOLT-LMM_manual.html'
+description = """The BOLT-LMM software package currently consists of two main algorithms, the BOLT-LMM algorithm for mixed model association testing, and the BOLT-REML algorithm for variance components analysis (i.e., partitioning of SNP-heritability and estimation of genetic correlations)."""
```
Comment (verdurin) [inline easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-2.3.4-foss-2020b.eb:7]: @btlindert please break this up into two lines

**Review by verdurin — CHANGES_REQUESTED** [review]

**Comment by verdurin** [comment]
@btlindert welcome to EasyBuild and thanks for the contribution. As you will have seen, I have made a couple of suggestions for improvement.

File: easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-fix-makefile.patch (hunk 1 above)
Comment (btlindert) [reply easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-fix-makefile.patch:1]: The patch updates the build arguments in the Makefile with are orginally for Intel. These can also be added to "buildopts" though, if preferred.

Comment (verdurin) [reply easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-fix-makefile.patch:1]: That's fine - please add the explanation as a comment inside the patch file.

File: easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-2.3.4-foss-2020b.eb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+
+source_urls = [
+    'https://storage.googleapis.com/broad-alkesgroup-public/%(name)s/downloads/',
+    'https://storage.googleapis.com/broad-alkesgroup-public/%(name)s/downloads/old/'
+]
+sources = ['%(name)s_v%(version)s.tar.gz']
+checksums = ['d7b587ea1a3cc950084c3cc8bfd9b8a75ca90899ee286261e9ea7a2c7f4c777a']
+patches = ['BOLT-LMM-fix-makefile.patch']
```
Comment (verdurin) [inline easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-2.3.4-foss-2020b.eb:20]: @btlindert please add the missing checksum for the patch.

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-2.3.4-foss-2020b.eb (hunk 3 above)
Comment (btlindert) [reply easybuild/easyconfigs/b/BOLT-LMM/BOLT-LMM-2.3.4-foss-2020b.eb:19]: @verdurin thanks for your comments. I've added the checksums and the patch explanation.

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

**Comment by verdurin** [comment]
Here's the error:
```
g++ -O2 -ftree-vectorize -march=native -fno-math-errno -O2 -msse -msse2 -DUSE_SSE -DMEASURE_DGEMM -DVERBOSE -Wall -fopenmp -I/users/boegelbot/CentOS8/haswell/software/Boost/1.74.0-GCC-10.2.0/include -I/users/boegelbot/CentOS8/haswell/software/NLopt/2.6.2-GCCcore-10.2.0/api -o memcpy.o -c memcpy.cpp
g++ -fopenmp -o bolt BoltMain.o Bolt.o BoltParams.o BoltParEstCV.o BoltReml.o CovariateBasis.o DataMatrix.o FileUtils.o Jackknife.o LDscoreCalibration.o MapInterpolater.o MatrixUtils.o MemoryUtils.o NonlinearOptMulti.o NumericUtils.o PhenoBuilder.o RestrictSnpSet.o SnpData.o SnpInfo.o SpectrumTools.o StatsUtils.o StringUtils.o Timer.o memcpy.o -L/users/boegelbot/CentOS8/haswell/software/Boost/1.74.0-GCC-10.2.0/lib -Wl,-rpath,/users/boegelbot/CentOS8/haswell/software/Boost/1.74.0-GCC-10.2.0/lib -L/users/boegelbot/CentOS8/haswell/software/NLopt/2.6.2-GCCcore-10.2.0/.libs -Wl,-rpath,/users/boegelbot/CentOS8/haswell/software/NLopt/2.6.2-GCCcore-10.2.0/.libs -lboost_program_options -lboost_iostreams -lz -lnlopt -lopenblas -llapack -lgfortran  -lpthread -lm 
/users/boegelbot/CentOS8/haswell/software/binutils/2.35-GCCcore-10.2.0/bin/ld.gold: error: cannot find -llapack
collect2: error: ld returned 1 exit status
make: *** [Makefile:127: bolt] Error 1
 (at easybuild/easybuild-framework/easybuild/tools/run.py:577 in parse_cmd_output)
 ```

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/356e71282a460d42b4ac280c3e32c456 for a full test report.

**Comment by boegel** [comment]
closing this since `foss/2020b` is deprecated, see https://docs.easybuild.io/policies/toolchains

**Pull request closed without merging**